dwave/optimization/include/dwave-optimization/array.hpp

```diff
@@ -231,20 +231,19 @@ struct Update {
     double value;   // The new/current value.
 };
 
-// Represents an Array
-//
-// This interface is designed to work with Python's buffer protocol
-// https://docs.python.org/3/c-api/buffer.html
-// We however, only implement a subset of the features. Specifically:
-//  * We only support doubles
-//
-// We also support a notion of state-dependent size. This allows nodes to
-// change their size based on the state of the decision variables.
-// Nodes are only permitted to grow and shrink along axis 0. This means that
-// growing or shrinking a node is equivalent to extending or shrinking the
-// buffer - there are no insertions needed.
-// Nodes will signal that they have a state-dependent size by returning negative
-// values for size() and in the first element of shape().
+/// An array.
+///
+/// This interface is designed to work with
+/// [Python's buffer protocol](https://docs.python.org/3/c-api/buffer.html).
+///
+/// We also support a notion of state-dependent size. This allows arrays to
+/// change their size based on the state of the decision variables.
+/// Arrays are only permitted to grow and shrink along axis 0. This means that
+/// growing or shrinking an array is equivalent to extending or shrinking the
+/// buffer - there are no insertions needed.
+/// Arrays will signal that they have a state-dependent size by returning negative
+/// values for Array::size() and in the first element of Array::shape(). Also,
+/// the Array::dynamic() method has been provided for convenience.
```

Comment:
```suggestion
/// Support for state-dependent size: arrays are allowed to change their size
/// based on the state of decision variables. Arrays are permitted to extend and
/// contract only along axis 0. Such operations are equivalent to growing or
/// shrinking a buffer, with no insertions being needed.
/// Arrays signal a state-dependent size by returning negative values for
/// Array::size() and in the first element of Array::shape(). For convenience,
/// the Array::dynamic() method is provided.
```

Reply:
"Support for state-dependent size: arrays are allowed to change their size based on the state of decision variables." <- this is quite unclear. I gather you object to the word "we"? I can reword it.

Reply:
Two things: no royal "we" and so it's clear from the start of this paragraph that it is all about dynamic size (in lieu of a header)

Reply:
In that document you referenced it shows support for MD headers too